src/utils.php

```diff
@@ -645,3 +645,87 @@ function ensure_dir( $dir ) {
 
 	return realpath( $dir );
 }
+
+/**
+ * Recursively check permissions in a directory and confirm the UID/GID is properly set.
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return string|bool Error message or true if all checks pass.
```

Comment:
I would refactor this function to return nothing and `exit` with an error if the machine is Linux and the file permissions are off.